### ISSUE TITLE
NATS tests do not work on the platform

### ISSUE THREAD

**Issue by ppalaga** [apache/camel-quarkus#1803]
This is caused by the fact that the tlsAuthContainer expects the certs to be accessible locally. This should be fixable by copying the files from ClassLoader.getResource() to the filesystem just before starting the container

**Comment by jamesnetherton** [comment]
Possibly related recent CI failure? If not we can raise another issue:

```
2020-09-18T13:51:08.0090196Z 2020-09-18 13:51:01,458 WARN  [org.tes.uti.RegistryAuthLocator] (main) Failure when attempting to lookup auth config (dockerImageName: testcontainersofficial/ryuk:0.3.0, configFile: /home/runner/.docker/config.json. Falling back to docker-java default behaviour. Exception message: /home/runner/.docker/config.json (No such file or directory)
2020-09-18T13:51:08.0097650Z 2020-09-18 13:51:02,274 INFO  [org.tes.DockerClientFactory] (main) Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
2020-09-18T13:51:08.0109901Z 2020-09-18 13:51:02,274 INFO  [org.tes.DockerClientFactory] (main) Checking the system...
2020-09-18T13:51:08.0117156Z 2020-09-18 13:51:02,275 INFO  [org.tes.DockerClientFactory] (main) ✔︎ Docker server version should be at least 1.6.0
2020-09-18T13:51:08.0151813Z 2020-09-18 13:51:02,413 INFO  [org.tes.DockerClientFactory] (main) ✔︎ Docker environment should have more than 2GB free disk space
2020-09-18T13:51:08.0179268Z 2020-09-18 13:51:02,437 INFO  [🐳 .1.4]] (main) Pulling docker image: nats:2.1.4. Please be patient; this may take some time but only needs to be done once.
2020-09-18T13:51:08.0185268Z 2020-09-18 13:51:02,731 INFO  [🐳 .1.4]] (tc-okhttp-stream-261160798) Starting to pull image
2020-09-18T13:51:08.0192527Z 2020-09-18 13:51:02,769 INFO  [🐳 .1.4]] (tc-okhttp-stream-261160798) Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
2020-09-18T13:51:08.0202406Z 2020-09-18 13:51:02,875 INFO  [🐳 .1.4]] (tc-okhttp-stream-261160798) Pulling image layers:  1 pending,  1 downloaded,  0 extracted, (477 bytes/? MB)
2020-09-18T13:51:08.0209298Z 2020-09-18 13:51:02,947 INFO  [🐳 .1.4]] (tc-okhttp-stream-261160798) Pulling image layers:  0 pending,  2 downloaded,  0 extracted, (40 KB/3 MB)
2020-09-18T13:51:08.0218576Z 2020-09-18 13:51:03,192 INFO  [🐳 .1.4]] (tc-okhttp-stream-261160798) Pulling image layers:  0 pending,  2 downloaded,  1 extracted, (3 MB/3 MB)
2020-09-18T13:51:08.0227592Z 2020-09-18 13:51:03,268 INFO  [🐳 .1.4]] (tc-okhttp-stream-261160798) Pulling image layers:  0 pending,  2 downloaded,  2 extracted, (3 MB/3 MB)
2020-09-18T13:51:08.0235221Z 2020-09-18 13:51:03,312 INFO  [🐳 .1.4]] (main) Creating container for image: nats:2.1.4
2020-09-18T13:51:08.0241831Z 2020-09-18 13:51:03,520 INFO  [🐳 .1.4]] (main) Starting container with ID: 3c004a2f5249d605d363a5efee212ea1fbf40eff995126b0e3f0d3a6138a0bb7
2020-09-18T13:51:08.0248754Z 2020-09-18 13:51:04,062 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 is starting: 3c004a2f5249d605d363a5efee212ea1fbf40eff995126b0e3f0d3a6138a0bb7
2020-09-18T13:51:08.0254133Z 2020-09-18 13:51:04,094 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 started in PT3.468S
2020-09-18T13:51:08.0322239Z 2020-09-18 13:51:04,095 INFO  [🐳 .1.4]] (main) Creating container for image: nats:2.1.4
2020-09-18T13:51:08.0328904Z 2020-09-18 13:51:04,116 INFO  [🐳 .1.4]] (main) Starting container with ID: 9b374c4f3100720031b11492f07977300348f7e319395975d7fc33c0a363352f
2020-09-18T13:51:08.0335267Z 2020-09-18 13:51:04,481 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 is starting: 9b374c4f3100720031b11492f07977300348f7e319395975d7fc33c0a363352f
2020-09-18T13:51:08.0340333Z 2020-09-18 13:51:05,089 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 started in PT0.994S
2020-09-18T13:51:08.0344976Z 2020-09-18 13:51:05,095 INFO  [🐳 .1.4]] (main) Creating container for image: nats:2.1.4
2020-09-18T13:51:08.0350788Z 2020-09-18 13:51:05,225 INFO  [🐳 .1.4]] (main) Starting container with ID: 00318aa9e463b778838dac9ea22cd2f7fefae34d4f644dc979d30bc489f62101
2020-09-18T13:51:08.0357759Z 2020-09-18 13:51:05,584 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 is starting: 00318aa9e463b778838dac9ea22cd2f7fefae34d4f644dc979d30bc489f62101
2020-09-18T13:51:08.0363545Z 2020-09-18 13:51:06,101 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 started in PT1.009S
2020-09-18T13:51:08.0368024Z 2020-09-18 13:51:06,102 INFO  [🐳 .1.4]] (main) Creating container for image: nats:2.1.4
2020-09-18T13:51:08.0373830Z 2020-09-18 13:51:06,134 INFO  [🐳 .1.4]] (main) Starting container with ID: ffa2c6ff530a8204b875901a79ad179587a6c5207795cc936c01ae7652775182
2020-09-18T13:51:08.0402198Z 2020-09-18 13:51:06,502 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 is starting: ffa2c6ff530a8204b875901a79ad179587a6c5207795cc936c01ae7652775182
2020-09-18T13:51:08.0404786Z 2020-09-18 13:51:07,107 INFO  [🐳 .1.4]] (main) Container nats:2.1.4 started in PT1.005S
2020-09-18T13:51:08.0859266Z 2020-09-18 13:51:08,084 INFO  [org.apa.cam.qua.cor.CamelBootstrapRecorder] (main) bootstrap runtime: org.apache.camel.quarkus.core.CamelContextRuntime
2020-09-18T13:51:08.2410225Z 2020-09-18 13:51:08,240 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main) Apache Camel 3.5.0 (camel-1) is starting
2020-09-18T13:51:08.2430589Z 2020-09-18 13:51:08,242 INFO  [org.apa.cam.imp.eng.AbstractCamelContext] (main) StreamCaching is not in use. If using streams then its recommended to enable stream caching. See more details at http://camel.apache.org/stream-caching.html
2020-09-18T13:51:08.2992154Z 2020-09-18 13:51:08,297 INFO  [org.apa.cam.imp.eng.InternalRouteStartupManager] (main) Route: basic-auth started and consuming from: natsBasicAuth://test
2020-09-18T13:51:08.3079473Z 2020-09-18 13:51:08,305 INFO  [org.apa.cam.imp.eng.InternalRouteStartupManager] (main) Route: no-auth started and consuming from: natsNoAuth://test
2020-09-18T13:51:08.3252922Z 2020-09-18 13:51:08,324 INFO  [org.apa.cam.imp.eng.InternalRouteStartupManager] (main) Route: token-auth started and consuming from: natsTokenAuth://test
2020-09-18T13:51:08.3280879Z 2020-09-18 13:51:08,326 INFO  [org.apa.cam.sup.jss.SSLContextParameters] (main) Available providers: SUN version 1.8.
2020-09-18T13:51:09.2801611Z 2020-09-18 13:51:09,277 ERROR [org.apa.cam.imp.eng.AbstractCamelContext] (main) Error starting CamelContext (camel-1) due to exception thrown: java.io.IOException: Unable to connect to NATS servers: nats://localhost:32840.: org.apache.camel.RuntimeCamelException: java.io.IOException: Unable to connect to NATS servers: nats://localhost:32840.
2020-09-18T13:51:09.2804910Z 	at org.apache.camel.RuntimeCamelException.wrapRuntimeException(RuntimeCamelException.java:66)
2020-09-18T13:51:09.2807058Z 	at org.apache.camel.support.service.BaseService.doFail(BaseService.java:409)
2020-09-18T13:51:09.2808833Z 	at org.apache.camel.support.service.BaseService.fail(BaseService.java:338)
2020-09-18T13:51:09.2810822Z 	at org.apache.camel.support.service.BaseService.start(BaseService.java:128)
2020-09-18T13:51:09.2813616Z 	at org.apache.camel.impl.engine.AbstractCamelContext.startService(AbstractCamelContext.java:3126)
2020-09-18T13:51:09.2817062Z 	at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRouteConsumers(InternalRouteStartupManager.java:362)
2020-09-18T13:51:09.2822202Z 	at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartRouteConsumers(InternalRouteStartupManager.java:282)
2020-09-18T13:51:09.2857506Z 	at org.apache.camel.impl.engine.InternalRouteStartupManager.safelyStartRouteServices(InternalRouteStartupManager.java:181)
2020-09-18T13:51:09.2861131Z 	at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRoutes(InternalRouteStartupManager.java:115)
2020-09-18T13:51:09.2864037Z 	at org.apache.camel.impl.engine.AbstractCamelContext.doStartCamel(AbstractCamelContext.java:2907)
2020-09-18T13:51:09.2866621Z 	at org.apache.camel.impl.engine.AbstractCamelContext.doStartContext(AbstractCamelContext.java:2740)
2020-09-18T13:51:09.2868873Z 	at org.apache.camel.impl.engine.AbstractCamelContext.doStart(AbstractCamelContext.java:2703)
2020-09-18T13:51:09.2870664Z 	at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
2020-09-18T13:51:09.2872599Z 	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2469)
2020-09-18T13:51:09.2874759Z 	at org.apache.camel.quarkus.core.CamelContextRuntime.start(CamelContextRuntime.java:57)
2020-09-18T13:51:09.2876968Z 	at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:45)
2020-09-18T13:51:09.2900711Z 	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy_0(CamelBootstrapProcessor$boot-173480958.zig:101)
2020-09-18T13:51:09.2902963Z 	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy(CamelBootstrapProcessor$boot-173480958.zig:40)
2020-09-18T13:51:09.2904589Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:565)
2020-09-18T13:51:09.2905727Z 	at io.quarkus.runtime.Application.start(Application.java:90)
2020-09-18T13:51:09.2906845Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-09-18T13:51:09.2908268Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-09-18T13:51:09.2910353Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-09-18T13:51:09.2911680Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-09-18T13:51:09.2912866Z 	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:212)
2020-09-18T13:51:09.2914606Z 	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:232)
2020-09-18T13:51:09.2916569Z 	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:504)
2020-09-18T13:51:09.2953460Z 	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:537)
2020-09-18T13:51:09.2956416Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$7(ClassBasedTestDescriptor.java:359)
2020-09-18T13:51:09.2960201Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-18T13:51:09.2963940Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:359)
2020-09-18T13:51:09.2967603Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:189)
2020-09-18T13:51:09.2970511Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
2020-09-18T13:51:09.2973207Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
2020-09-18T13:51:09.2976737Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-18T13:51:09.2979524Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-09-18T13:51:09.2981725Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-09-18T13:51:09.2983938Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-09-18T13:51:09.2986724Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-18T13:51:09.2989779Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-09-18T13:51:09.3100797Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-09-18T13:51:09.3111285Z 	at java.util.ArrayList.forEach(ArrayList.java:1259)
2020-09-18T13:51:09.3115624Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-09-18T13:51:09.3124248Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-09-18T13:51:09.3126955Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-18T13:51:09.3129930Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-09-18T13:51:09.3132222Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-09-18T13:51:09.3134412Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-09-18T13:51:09.3140952Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-09-18T13:51:09.3345027Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-09-18T13:51:09.3347728Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-09-18T13:51:09.3351933Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2020-09-18T13:51:09.3356290Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2020-09-18T13:51:09.3359916Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2020-09-18T13:51:09.3364063Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
2020-09-18T13:51:09.3365863Z 	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
2020-09-18T13:51:09.3368059Z 	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
2020-09-18T13:51:09.3370345Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
2020-09-18T13:51:09.3372840Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
2020-09-18T13:51:09.3375891Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
2020-09-18T13:51:09.3378724Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
2020-09-18T13:51:09.3381469Z 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
2020-09-18T13:51:09.3385383Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
2020-09-18T13:51:09.3387253Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
2020-09-18T13:51:09.3388809Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
2020-09-18T13:51:09.3389996Z Caused by: java.io.IOException: Unable to connect to NATS servers: nats://localhost:32840.
2020-09-18T13:51:09.3391202Z 	at io.nats.client.impl.NatsConnection.connect(NatsConnection.java:248)
2020-09-18T13:51:09.3392457Z 	at io.nats.client.impl.NatsImpl.createConnection(NatsImpl.java:29)
2020-09-18T13:51:09.3393898Z 	at io.nats.client.Nats.createConnection(Nats.java:271)
2020-09-18T13:51:09.3394713Z 	at io.nats.client.Nats.connect(Nats.java:186)
2020-09-18T13:51:09.3396418Z 	at org.apache.camel.component.nats.NatsEndpoint.getConnection(NatsEndpoint.java:80)
2020-09-18T13:51:09.3617464Z 	at org.apache.camel.component.nats.NatsConsumer.doStart(NatsConsumer.java:62)
2020-09-18T13:51:09.3619243Z 	at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
2020-09-18T13:51:09.3620443Z 	... 61 more
```

**Comment by ppalaga** [comment]
Is that an intermittent failure? The NATS tests are passing for me locally.

**Comment by ppalaga** [comment]
This issue is a false alarm. As of current Camel Quarkus master, the NATS test works in the platform. Sorry for the noise.

@jamesnetherton please file a separate issue for the mentioned failure if necessary.
